### PR TITLE
Do not throw strings

### PR DESCRIPTION
**Pull request by Apollon77**

### CODE REVIEW

**Comment by Apollon77** [comment]
Change to a typed EndOfStreamError

**Comment by Apollon77** [comment]
@mfucci Seems we missed to click the ok button here ... Adjustments done, Ready for re-review

File: src/util/Stream.ts [hunk 1]
```diff
@@ -3,9 +3,16 @@
  * Copyright 2022 The node-matter Authors
  * SPDX-License-Identifier: Apache-2.0
  */
+import { MatterError } from "../error/MatterError";
 
 export const END_OF_STREAM = "End-of-Stream";
```
Comment (mfucci) [inline src/util/Stream.ts:8]: You can delete this

File: src/util/Stream.ts [hunk 2]
```diff
@@ -3,9 +3,16 @@
...
 
+export class EndOfStreamError extends MatterError {
+    constructor() {
```
Comment (mfucci) [inline src/util/Stream.ts:11]: You don't need a constructor, the class type is enough.

**Review by mfucci — CHANGES_REQUESTED** [review]

**Comment by Apollon77** [comment]
Adjusted

**Review by mfucci — APPROVED** [review]

**Pull request merged**